Задача по 12 спринту

Alcesta opened the pull request:
Задача по 12 спинту

Commit:
Задача по 12 спринту

Commit:
правки по ревью
задача по 12 спринту

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListCell.swift:
```diff
@@ -1,9 +1,30 @@
+   
     @IBOutlet var cellImage: UIImageView!
     @IBOutlet var likeButton: UIButton!
     @IBOutlet var dateLabel: UILabel!
```
и всё таки лучше сделать их приватными, по моей рекомендации из прошлого спринта(

Alcesta replied:
если я делаю их приватными я не могу их использоваться из ImagesListViewController где они у меня вызываются

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListCell.swift:
```diff
@@ -1,9 +1,30 @@
+        cellImage.kf.cancelDownloadTask()
+    }
+    
+    @IBAction func likeButtonClicked(_ sender: Any) {
```
Ну а тут точно нужен private, IBAction должны быть приватными всегда

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -19,26 +22,41 @@ final class ImagesListViewController: UIViewController {
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         if segue.identifier == showSingleImageSegueIdentifier {
             let viewController = segue.destination as! SingleImageViewController
             let indexPath = sender as! IndexPath
```
В этих строках есть force операция - force cast, она как и force unwrap является нежелательной и надо стараться такого кода избегать. Можно переписать без их использования, переместив в мягкий каст в if: if let a = a as? 
Всегда когда есть возможность написать безопасную конструкцию – отдавай предпочтение ей

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -19,26 +22,41 @@ final class ImagesListViewController: UIViewController {
-        cell.cellImage.image = image
-        cell.dateLabel.text = dateFormatter.string(from: Date())
-        let isLikeButton = indexPath.row % 2 != 0
+        let photo = photos[indexPath.row]
```
старайся использовать safe при работе с индексами: 
let photo = photos[safe: indexPath.row]

Мало ли что, вдруг по индексу элемента не будет, а прила упадет

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -81,3 +101,40 @@ extension ImagesListViewController: UITableViewDelegate {
+                cell.setIsLiked(isLikeButton: self.photos[indexPath.row].isLiked)
+                UIBlockingProgressHUD.dismiss()
+            case .failure(let error):
+                UIBlockingProgressHUD.dismiss()
```
Дублирование UIBlockingProgressHUD. Его можно вывести за пределы switch, так как его вызывать в любом случае)

andrew-kuzn-yndx commented on ImageFeed/Profile/ProfileImageService.swift:
```diff
@@ -8,32 +8,28 @@ final class ProfileImageService {
             case .success(let userResult):
                 self.avatarURL = userResult.profileImage["large"]
-                completionOnMainThread(.success(self.avatarURL!))
+                completion(.success(self.avatarURL!))
```
стоит поправить это и не использовать force unwrap

Commit:
Поправила  замечания после ревью
Хочу заметить, что кнопка назад к списку картинок у меня реализована и работает https://jing.yandex-team.ru/files/ilushina/photo_2024-07-21%2017.59.11.d162bfa.jpeg тут доказательства могу показать видео
устранила нежелательный распаковки
где-то поставила приватность, где-то добавила weak
стараюсь успеть в жесткий дедлайн

Commit:
косметические правки в ProfileImageService

Commit:
По замечанию ревьюера удалила и заново добавила иконку назад, чтобы заработало и у них

Commit:
По замечанию ревьюера удалила и заново добавила иконку кнопки Назад

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -126,13 +133,14 @@ extension ImagesListViewController: ImagesListCellDelegate {
             guard let self = self else { return }
+            defer {
+                UIBlockingProgressHUD.dismiss()
+                       }
```
👍
